server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -342,14 +343,14 @@ private boolean mergeParam(RuleParamDto paramDto, RulesDefinition.Param paramDef
     return changed;
   }
 
-  private boolean mergeTags(RulesDefinition.Rule ruleDef, RuleDto dto) {
+  private static boolean mergeTags(RulesDefinition.Rule ruleDef, RuleDto dto) {
     boolean changed = false;
 
     if (RuleStatus.REMOVED == ruleDef.status()) {
       dto.setSystemTags(Collections.<String>emptySet());
       changed = true;
     } else if (!dto.getSystemTags().containsAll(ruleDef.tags())
-      || !Sets.intersection(dto.getTags(), ruleDef.tags()).isEmpty()) {
+      || !intersection(dto.getTags(), ruleDef.tags()).isEmpty()) {
       dto.setSystemTags(ruleDef.tags());
```

Comment:
system tags = tags provided by plugin => cannot be changed by user
other tag = user tags